File: app/src/main/java/example/jllarraz/com/passportreader/ui/fragments/PassportPhotoFragment.java
package example.jllarraz.com.passportreader.ui.fragments;

import android.content.Context;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import example.jllarraz.com.passportreader.R;
import example.jllarraz.com.passportreader.common.IntentData;
import example.jllarraz.com.passportreader.data.Passport;
import example.jllarraz.com.passportreader.ui.views.TouchImageView;

public class PassportPhotoFragment extends Fragment{

    private PassportPhotoFragmentListener passportPhotoFragmentListener;

    private Bitmap bitmap;

    private TouchImageView appCompatImageViewFace;

    public static PassportPhotoFragment newInstance(Bitmap bitmap) {
        PassportPhotoFragment myFragment = new PassportPhotoFragment();
        Bundle args = new Bundle();
        args.putParcelable(IntentData.KEY_IMAGE, bitmap);
        myFragment.setArguments(args);
        return myFragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View inflatedView = inflater.inflate(R.layout.fragment_photo, container, false);

        Bundle arguments = getArguments();
        if(arguments.containsKey(IntentData.KEY_IMAGE)){
            bitmap = (Bitmap) arguments.getParcelable(IntentData.KEY_IMAGE);
        } else {
            //error
        }

        appCompatImageViewFace =  inflatedView.findViewById(R.id.image);

        return inflatedView;
    }

    @Override
    public void onResume() {
        super.onResume();
        refreshData(bitmap);
    }

    private void refreshData( Bitmap bitmap){
        if(bitmap == null){
            return;
        }
        appCompatImageViewFace.setImageBitmap(bitmap);
    }


    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        FragmentActivity activity = getActivity();
        if(activity instanceof PassportPhotoFragmentListener){
            passportPhotoFragmentListener = (PassportPhotoFragmentListener) activity;
        }
    }

    @Override
    public void onDetach() {
        passportPhotoFragmentListener = null;
        super.onDetach();

    }

    public interface PassportPhotoFragmentListener {

    }

}

File: app/src/main/java/example/jllarraz/com/passportreader/ui/fragments/PassportPhotoFragment.kt
package example.jllarraz.com.passportreader.ui.fragments

import android.content.Context
import android.graphics.Bitmap
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import example.jllarraz.com.passportreader.R
import example.jllarraz.com.passportreader.common.IntentData
import example.jllarraz.com.passportreader.databinding.FragmentPhotoBinding

class PassportPhotoFragment : Fragment(R.layout.fragment_photo) {

    private var passportPhotoFragmentListener: PassportPhotoFragmentListener? = null

    private var bitmap: Bitmap? = null

    private lateinit var binding: FragmentPhotoBinding
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding = FragmentPhotoBinding.bind(view)
        val arguments = arguments
        if (arguments!!.containsKey(IntentData.KEY_IMAGE)) {
            bitmap = arguments.getParcelable<Bitmap>(IntentData.KEY_IMAGE)
        }
    }

    override fun onResume() {
        super.onResume()
        refreshData(bitmap)
    }

    private fun refreshData(bitmap: Bitmap?) {
        if (bitmap == null) {
            return
        }
        binding.image.setImageBitmap(bitmap)
    }


    override fun onAttach(context: Context) {
        super.onAttach(context)
        val activity = activity
        if (activity is PassportPhotoFragmentListener) {
            passportPhotoFragmentListener = activity
        }
    }

    override fun onDetach() {
        passportPhotoFragmentListener = null
        super.onDetach()

    }

    interface PassportPhotoFragmentListener

    companion object {

        fun newInstance(bitmap: Bitmap): PassportPhotoFragment {
            val myFragment = PassportPhotoFragment()
            val args = Bundle()
            args.putParcelable(IntentData.KEY_IMAGE, bitmap)
            myFragment.arguments = args
            return myFragment
        }
    }

}
